File: python/problem005.py
prime_numbers = [2]
max_values = []
temp_values = []


def is_prime(n):
    for prime in prime_numbers:
        if n % prime == 0:
            return False
    return True


def init_temp_values():
    for a in range(0, len(temp_values)):
        temp_values[a] = 0


for i in range(3, 21):
    if is_prime(i):
        prime_numbers.append(i)

for a in prime_numbers:
    max_values.append(0)
    temp_values.append(0)

for x in range(2, 21):
    init_temp_values()
    number = x
    pointer = 0
    while number > 1:
        if number % prime_numbers[pointer] == 0:
            number /= prime_numbers[pointer]
            temp_values[pointer] += 1
        else:
            pointer += 1
    for y in range(0, len(temp_values)):
        if max_values[y] < temp_values[y]:
            max_values[y] = temp_values[y]

result = 1
for i in range(0, len(max_values)):
    result *= prime_numbers[i] ** max_values[i]

print(result)

File: javascript/problem005.js
var primeNumbers = [2];
var maxValues = [];
var tempValues = [];
function isPrime(n){

  for(a in primeNumbers){
    if(n % primeNumbers[a] == 0){
      return false;
    }
  }
  return true;
}

for(i=3; i<=20; i++){
  if(isPrime(i)){
    primeNumbers.push(i);
  };
}

for(a in primeNumbers){
  maxValues.push(0);
  tempValues.push(0);
}

function initTempValues(){
  for(a in tempValues){
    tempValues[a] = 0;
  }
}

for(x=2; x<=20; x++){
  initTempValues();
  var number = x;
  var pointer = 0;
  while(number > 1){
    if(number % primeNumbers[pointer] == 0){
      number = number / primeNumbers[pointer];
      tempValues[pointer] = tempValues[pointer] + 1;
    }
    else{
      pointer = pointer + 1;
    }
  }
  for(y = 0; y < tempValues.length; y++){
    if(maxValues[y] < tempValues[y]){
      maxValues[y] = tempValues[y];
    }
  }
}

function square(x, y){
  var value = 1;
  for(a=0; a<y; a++){
    value = value * x;
  }
  return value;
}

var result = 1;
for(i=0; i<maxValues.length; i++){
  result = result * square(primeNumbers[i], maxValues[i]);
}
console.log(result);
